File: Python/sumOfPairs.py
# s = 6
# ints = [4, 3, 2, 3, 4]

ints = [10, 5, 2, 3, 7, 5]
s = 10


cache = set()
for i in ints:
    print(i)
    if s - i in cache:
        print([s - i, i])
    cache.add(i)


print(cache)

File: Rust/5 Kyu/sum_of_pairs.rs
fn sum_pairs(ints: &[i8], s: i8) -> Option<(i8, i8)> {
    let mut copy = vec![];
    let mut cont:usize = 0;
    let tam = ints.len();
    while cont<tam{
        let atual:i8 = ints[cont];
        let  diff = s-atual;
        if copy.contains(&diff){
            return Some((diff,atual));
        }
        if !copy.contains(&atual){
            copy.push(atual);
        }
        cont+=1;
    }
    
    None
    
    
    }